### ISSUE TITLE
Allow for more flexibility streaming data into NodeIO from various cloud locations 

### ISSUE THREAD

**Issue by leefsmp** [donmccurdy/glTF-Transform#480]
Thanks for the package, it's a great help. We are using it to load a gltf file server-side in an AWS Lambda function in order to do some preprocessing on our models.

For that purpose I stream the model directly from S3 to avoid downloading all the files locally on the lambda disk. In order to achieve that, I didn't find a way to use signed urls as each file of the .gltf would need to be downloaded using different query parameters in their url. I found a nice way to do that using that package: [fetch-hooks](https://www.npmjs.com/package/fetch-hooks). 

However it requires the uri to be in the form `s3://bucket/object...` and not `http|https`.

What I did is to slightly modify the code in your package to allow uri's starting with `s3://` to be recognized as network locations, so basically:
```
export class HTTPUtils {
  ...
  static readonly S3_REGEXP = /s3?:\/\//;

  static isAbsoluteURL(path: string) {
    return this.S3_REGEXP.test(path);
  }
  
  ...
```

this just works great, I am able to stream my models directly from S3 bucket inside the lambda as follow:

```
import { NodeIO } from '@gltf-transform/core';
const { hook, fetch, hooks } = require('fetch-hooks')
const AWS = require('aws-sdk')
const S3 = new AWS.S3()

const s3hook = hooks.s3(S3, { 
    baseURI: `s3://bucket`, 
    acl: 'private' 
});

const fetchHook = hook(fetch, s3hook);

const io = new NodeIO(fetchHook).setAllowHTTP(true)

const document = await io.read(`s3://bucket/assetId/model.gltf`);	
```

That would be great if NodeIO allows for such customization outside the box by providing some method or exposing the internal `HTTPUtils`. Maybe a way to pass a custom HTTPUtils object to handle such scenario.

Thanks

**Comment by donmccurdy** [comment]
Hi @leefsmp! I'd be okay with the `isAbsoluteURL` method returning true for all protocols, not just HTTP/HTTPS. Is that all this would require, or does NodeIO need to treat the `fetch-hooks` package differently somehow?

Also note that a custom I/O class can be extended from NodeIO or PlatformIO pretty easily for more custom behavior, see DenoIO for a simpler example: https://github.com/donmccurdy/glTF-Transform/blob/76f001b/packages/core/src/io/deno-io.ts

**Comment by leefsmp** [comment]
It would just work if `isAbsoluteURL` returns true for all protocols that's an easier approach indeed. 

Thanks for pointing out the custom IO implementation, however in my case I would need to override each method of NodeIO that contains a call to `isAbsoluteURL`, a bit of an overkill. 

Thanks.

**Comment by donmccurdy** [comment]
Thanks! To be fixed in #485.
